src/cameras/video_writer/video_writer.py

```diff
@@ -0,0 +1,55 @@
+import logging
+import os
+import traceback
+from pathlib import Path
+from typing import List
+
+import cv2
+from pydantic import BaseModel
+
+from src.cameras.dto import FramePayload
+
+logger = logging.getLogger(__name__)
+
+
+class SaveOptions(BaseModel):
+    writer_dir: Path
+    filename: str = "movie.mp4"
+    fps: float
+    fourcc: str = "MP4V"
+    frame_width: int
+    frame_height: int
+
+    @property
+    def full_path(self):
+        return Path().joinpath(self.writer_dir, self.filename)
+
+
+class VideoWriter:
```

Comment:
The VideoWriter class simply keeps track of frames in an array. When "Save" is called, it safely writes all frame data to a file using the cv2.VideoWriter class, and then releases the resource.